### ISSUE TITLE
perf: don't save the packuments to cache (unless `prefer-offline` is set to `true`)

### ISSUE THREAD

**Issue by zkochan** [pnpm/pnpm#7260]
### Contribution

- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

Right now pnpm fetches the packument of the resolved package during every installation (unless the package is specified via exact version). Hence, it doesn't really make a lot of sense to save these packument files in the cache (aka metadata cache).

### Describe the solution you'd like

I suggest to save the package metadata files to the cache only when the `prefer-offline` setting is set to `true`

### Describe the drawbacks of your solution

If there will be problems with internet connection, pnpm will not be able to run with `pnpm install --offline` as the cache will not contain any metadata files.

### Describe alternatives you've considered

-

**Comment by KSXGitHub** [comment]
I don't think this is going to have a big impact on performance because network request is always going to be an order of magnitude slower than local disk I/O.

Maybe you can add a `--no-packument-cache` flag, but that's going to add to pnpm complexity for a feature that I think most people don't care about (they already have a lockfile most of the time).

**Comment by zkochan** [comment]
This targets a part of pnpm that no longer exists, so closing.

This is about the metadata cache of the Node.js implementation. The resolver is native code now and caches package metadata differently.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
